src/Aspire.Hosting/Dcp/DcpHostService.cs

```diff
@@ -85,19 +85,19 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
         }
 
         _shutdownCts.Cancel();
+        await _appExecutor.StopAsync(cancellationToken).ConfigureAwait(false);
 
         await TaskHelpers.WaitIgnoreCancelAsync(_logProcessorTask, _logger, "Error in logging socket processor.").ConfigureAwait(false);
     }
 
     public async ValueTask DisposeAsync()
     {
-        if (_dcpRunDisposable is null)
+        if (_dcpRunDisposable is { } disposable)
         {
-            return;
-        }
+            _dcpRunDisposable = null;
 
-        await _dcpRunDisposable.DisposeAsync().ConfigureAwait(false);
-        _dcpRunDisposable = null;
+            await disposable.DisposeAsync().AsTask().ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
```

Comment:
Should errors here be logged? (ignoring OCE)

IMO try/catch with log would help prevent errors from being ignored.